src/transformers/commands/serving.py

```diff
@@ -12,32 +12,55 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import json
+import time
 from argparse import ArgumentParser, Namespace
-from typing import Any, List, Optional
+from dataclasses import dataclass, field
+from logging import getLevelNamesMapping
+from threading import Thread
+from typing import Any, Dict, Optional
 
-from ..pipelines import Pipeline, get_supported_tasks, pipeline
-from ..utils import logging
+from transformers.utils.import_utils import is_fastapi_available, is_pydantic_available, is_uvicorn_available
+
+from .. import PreTrainedTokenizerFast, TextIteratorStreamer
+from ..generation.continuous_batching import ContinuousBatchingManager, RequestStatus
+from ..utils import is_torch_available, logging
 from . import BaseTransformersCLICommand
 
 
-try:
-    from fastapi import Body, FastAPI, HTTPException
-    from fastapi.routing import APIRoute
-    from pydantic import BaseModel
-    from starlette.responses import JSONResponse
-    from uvicorn import run
+if is_torch_available():
+    import torch
 
-    _serve_dependencies_installed = True
-except (ImportError, AttributeError):
-    BaseModel = object
+    from transformers import (
+        AutoModelForCausalLM,
+        AutoTokenizer,
+        BitsAndBytesConfig,
+        GenerationConfig,
+        PreTrainedModel,
+    )
 
-    def Body(*x, **y):
-        pass
 
-    _serve_dependencies_installed = False
+if is_pydantic_available():
```

Comment:
(nit) might be worth checking pydantic + fastapi + uvicorn since we are importing from those 3

Reply:
or add a `transformers[serving]` extra in `setup.py` and check it here with `is_transformers_serving_available` that checks all 3